generator/osm2type.cpp

```diff
@@ -529,6 +529,8 @@ namespace ftype
           { "bicycle", "!", [&params] { params.AddType(types.Get(CachedTypes::NOBICYCLE)); }},
           { "bicycle", "~", [&params] { params.AddType(types.Get(CachedTypes::YESBICYCLE)); }},
           { "cycleway", "~", [&params] { params.AddType(types.Get(CachedTypes::YESBICYCLE)); }},
+          { "cycleway:right", "~", [&params] { params.AddType(types.Get(CachedTypes::YESBICYCLE)); }},
```

Comment:
Not sure I understand if you mean the road for cars is oneway, but that is not the case
You can for example add
cycleway:left=track
cycleway:right=lane
if the bicycles have some space on the right side of the line in OSM and a dedicated separated way on the left side of the OSM way, but that does not imply anything for the direction cars can go